### ISSUE TITLE
Cancel button is not highlighted when the cursor is brought near to it

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22658]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to settings > Workspace > choose workspace with a currency outside USD
2. Go to bank account
4. "Workspace currency " page will pop out
## Expected Result: 
The cancel button should be high lightened when the cursor is brought near to it
## Actual Result: 
The cancel button was not high lightened when the cursor is brought near to it
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.39-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/d97209a0-afe2-444e-9060-e930bb93d220

https://github.com/Expensify/App/assets/43996225/12836bb1-f698-4fe1-9d80-adb4e008aaa8

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Yokabdk
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689038554734879

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconsistency - Cancel button is not highlighted when the cursor is brought near to it.
### What is the root cause of that problem?
the Button has hover style only if it is has prop danger or success.

https://github.com/Expensify/App/blob/e5b7e3a767fcf70498c894c9e5f0f4d190e91fb0/src/components/Button/index.js#L297-L300


### What changes do you think we should make in order to solve the problem?
we need to add default hover style if button if not succuss or danger.
```javascript
    hoverStyle={[
        this.props.success && !this.props.isDisabled ? styles.buttonSuccessHovered : undefined,
        this.props.danger && !this.props.isDisabled ? styles.buttonDangerHovered : undefined,
        !this.props.danger && !this.props.success && !this.props.isDisabled ? styles.hoveredButton : undefined,
    ]}
```

### What alternative solutions did you explore? (Optional)
N/A

**Comment by DanutGavrus** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconsistency - Cancel button is not highlighted when the cursor is brought near to it

### What is the root cause of that problem?
In our custom `Button` component, we have hover styles only for `success` & `danger` buttons:
https://github.com/Expensify/App/blob/38890e2e3c32b52dad11a7dfc09494924519946b/src/components/Button/index.js#L297-L300
Our `Button` from the "Workspace currency" Modal gets neither `success` nor `danger` prop.

### What changes do you think we should make in order to solve the problem?
Apply the already existing `styles.hoveredButton` to the Button when it's neither `success` nor `danger` such as:
```js
hoverStyle={[
    this.props.success && !this.props.isDisabled ? styles.buttonSuccessHovered : undefined,
    this.props.danger && !this.props.isDisabled ? styles.buttonDangerHovered : undefined,
    !this.props.success && !this.props.danger && !this.props.isDisabled ? styles.hoveredButton : undefined,
]}
```
I think it's a good option to apply this to *all buttons which are not success nor danger* as a *default hover*. `styles.hoveredButton` uses the `backgroundColor: themeColors.buttonHoveredBG` which is `colors.greenDefaultButtonHover`.  

I see that the `buttonSuccessHovered` and `buttonDangerHovered` styles also set `borderWidth: 0` so we may create a *new default style* to apply this too if needed.

### What alternative solutions did you explore?
As an alternative, we may create a *new style* and apply it only to this `Cancel` button.

**Comment by conorpendergrast** [comment]
As noted [via Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1689047775281619?thread_ts=1689038554.734879&cid=C049HHMV9SM):
> Same behavior in **Remove members** and **Delete workspace** as well. Let's have a solution that fixed those too

**Comment by DanutGavrus** [comment]
@conorpendergrast Updated [alternative solution](https://github.com/Expensify/App/issues/22658#issuecomment-1631246684) in order to trigger only the 3 buttons mentioned.

**Comment by mollfpr** [comment]
Sorry, I got a problem with updating the NPM. I'll start the review in the morning.

**Comment by mollfpr** [comment]
@conorpendergrast Do we need a solution to change all the normal button hover styles or just specific for this cancel button on the workspace delete modal and remove workspace members modal? If we want to change the hover style for all the buttons (which is not a `success` or `danger` button), we should get approval from the design team.

---

Why do we need to check if it's not a success or danger? Can we just put it higher than the success and danger style and then let it overrides?

@dukenv0307 Is `borderWidth: 0` will have a visible effect on non success and danger buttons?

**Comment by mollfpr** [comment]
Bump @conorpendergrast for clarification [here](https://github.com/Expensify/App/issues/22658#issuecomment-1634140241)

**Comment by conorpendergrast** [comment]
I think we should only change the cancel button for this Bank account modal, and for **Delete workspace** and **Remove members**

**Comment by mollfpr** [comment]
Thanks @conorpendergrast!

It looks like @dukenv0307 [alternative solution](https://github.com/Expensify/App/issues/22658#issuecomment-1631246439) is what we need. I think changing all the normal button hover styles might lead to a regression, so having a prop to make it optional seems reasonable and avoids unexpected results for other buttons.

🎀 👀 🎀 C+ reviewed!

**Comment by DanutGavrus** [comment]
@mollfpr If we proceed to only change these 3 buttons, I think my alternative solution was the first one to propose that. From *12 July*:
> As an alternative, we may create a new style and apply it only to this Cancel button, to the correspomding remove members button and to the corresponding delete workspace one.

Duken's alternative was added on *18 July*.

**Comment by mollfpr** [comment]
@DanutGavrus Yes, you have a general idea for the chosen solution. I believe your alternative solution is based on your main solution `!this.props.success && !this.props.danger && !this.props.isDisabled ? styles.hoveredButton : undefined,` where I have asked it before do we really need to do that?

>Why do we need to check if it's not a success or danger? Can we just put it higher than the success and danger style and then let it get overridden?

So after that question, I will need more details and solutions that follow the expectation. @dukenv0307 did following the conversation and updated with a detailed solution.

@DanutGavrus Let me know if this not satisfied you, and we can ask to @conorpendergrast and @Gonals opinions about this.

**Comment by DanutGavrus** [comment]
@mollfpr From what I know, solutions & alternatives do not have to provide the full implementation, but only to convey the idea, which is the same one here. I've said:
> As an alternative, we may create a new style and apply it only to this Cancel button, to the corresponding remove members button and to the corresponding delete workspace one.  

From his alternative, `buttonNormalHovered` is a new style which is applied only to those 3 buttons through props, which, for me, seems to be the same **concept**. Applying the *new style* at the begging or the end will have the same effect as it's chosen conditionally, and it's position could also be handled in the PR.

The contributing guide states that the difference between solutions should be *important, meaningful or considerable*. As an example, on [this issue](https://github.com/Expensify/App/issues/18912) the reviewer chose the first Proposal and [said this](https://github.com/Expensify/App/issues/18912#issuecomment-1549563413):
> [...] I have picked the first one who posted the **concept. As for the implementation**, I believe all cases will be handled in the PR [...]

Another example would be [this issue](https://github.com/Expensify/App/issues/20624) where a [Proposal](https://github.com/Expensify/App/issues/20624#issuecomment-1617125332) was actually "a dupe" of my [alternative Proposal](https://github.com/Expensify/App/issues/20624#issuecomment-1590023343), but provided the full implementation, and we went with mine after clarifications as it conveyed the [same idea](https://github.com/Expensify/App/issues/20624#issuecomment-1630923136). For me, it seems to be the same case here, but let me know your feedback about these 😁.

**Comment by conorpendergrast** [comment]
@Gonals What's your take here?

**Comment by Gonals** [comment]
That was a bit hard to follow with all the edits, but ultimately, I agree that @DanutGavrus original alternative is pretty much the same as the selected solution. It is true it does not provide full implementation, but their main solution does and the changes would be minimal.

@mollfpr, I see you did ask for clarification on the if/else. Do you think that change is relevant enough to change or decision of the correct solution of would it be more of a review comment of a valid solution?

**Comment by DanutGavrus** [comment]
@dukenv0307
> But the way to apply it to these button doesn't mention in both main and alternative solution.  

I didn't explicitly mention how as I thought using props is straight forward & automatically thought of for passing props to set custom behavior to child components.

> in the further if we want to add hover style for other normal buttons, we can implement easily by adding this prop.  
1. It seems to be something extra, out of context of this issue;
2. And that will also be possible with my alternative solution, I think you shouldn't automatically imply it won't. I've said:
> [...] create a new style and apply it only to [...]  

I've never said we won't be able to apply it in other places. If my alternative solution makes it possible to choose where to apply it, I think it's implicit that it will be applicable in other places too.

**Comment by mollfpr** [comment]
>That was a bit hard to follow with all the edits, but ultimately, I agree that @DanutGavrus original alternative is pretty much the same as the selected solution. It is true it does not provide full implementation, but their main solution does and the changes would be minimal.

This is my mistake, I missed reviewing the alternative solution from @DanutGavrus. I agree that the idea is the same.

>@mollfpr, I see you did ask for clarification on the if/else. Do you think that change is relevant enough to change or decision of the correct solution of would it be more of a review comment of a valid solution?

This should not change anything, and this can be done in the PR review since I'm the one who asks for it. It will be different if the contributor suggests the expected solution beforehand.

>Note: Before submitting a proposal on an issue, be sure to read any other existing proposals. ALL NEW PROPOSALS MUST BE DIFFERENT FROM EXISTING PROPOSALS. The difference should be important, meaningful or considerable.

On the attached link on another issue, we follow the rules and assign the first contributor to propose it. Also, the expected solution should not be complicated to implement the same idea differently. So I believe the @DanutGavrus is entitled to get assigned here.

Sorry @dukenv0307 and @DanutGavrus for the confusions here 🙏

**Comment by DanutGavrus** [comment]
@conorpendergrast @Gonals Could you please assign? 😁

**Comment by DanutGavrus** [comment]
@mollfpr
PR ready for review: [PR 23464](https://github.com/Expensify/App/pull/23464)

**Comment by mollfpr** [comment]
@dukenv0307 Yes, we should—especially the cancel button alongside with success/danger button.

**Comment by DanutGavrus** [comment]
@mollfpr
As per what Conor [said above](https://github.com/Expensify/App/issues/22658#issuecomment-1638893769):
> I think we should only change the cancel button for this Bank account modal, and for Delete workspace and Remove members

https://github.com/Expensify/App/issues/23196 can also be fixed in the PR as per https://github.com/Expensify/App/issues/23196#issuecomment-1647657820

**Comment by mollfpr** [comment]
>@mollfpr DanutGavrus's https://github.com/Expensify/App/issues/22658#issuecomment-1631246684 just fix three cancel button while my https://github.com/Expensify/App/issues/22658#issuecomment-1631246439 fix for all cancel button.

@dukenv0307 I'm sorry, which part of your proposal do you mean?

I agree https://github.com/Expensify/App/issues/23196 can be fixed in the PR. I also will inspect the other button that needs to add the hover style while reviewing the PR.

**Comment by DanutGavrus** [comment]
Also, as Yokabdk [said](https://github.com/Expensify/App/issues/23196#issuecomment-1643931384):
> The issue https://github.com/Expensify/App/issues/22658 is only about cancel button for Bank account modal, and for Delete workspace and Remove members. This issue is about the 'Edit' Button. You can see the discussion for better clarification.

This issue is about 'Cancel' buttons, while that issue is about an 'Edit' button.

**Comment by mollfpr** [comment]
I think that's just a small detail. As I said before, I planned to evaluate whether the button needs to be added to the hover style. Eventually, any contributor working on this fix must update the other button.

**Comment by mollfpr** [comment]
@dukenv0307 I'm recommending you wait to open a PR/draft before you get assigned to the issue. Ultimately, both proposals' ideas are the same, and the implementation will be reviewed in the PR.

**Comment by mollfpr** [comment]
@dukenv0307 Thank you for your understanding. Once again, I'm sorry to give you confusion.

**Comment by zanyrenney** [comment]
This seems like a very similar issue - https://github.com/Expensify/App/issues/23196 @conorpendergrast do you think we should tackle these in one?

**Comment by mollfpr** [comment]
@zanyrenney We can put that in the PR for compliment since we only need one line code.

**Comment by zanyrenney** [comment]
Awesome thank you @mollfpr

**Comment by DanutGavrus** [comment]
@conorpendergrast Applied!

**Comment by Yokabdk** [comment]
@conorpendergrast I have applied using the upwork profile link https://www.upwork.com/freelancers/~01675be5906ac754ec so that I could get the payment easily. sorry for the delay

**Comment by Yokabdk** [comment]
@conorpendergrast I have applied using the upwork link https://www.upwork.com/freelancers/~01675be5906ac754ec so that I could get the payment easily. sorry for the delay

**Comment by conorpendergrast** [comment]
@Yokabdk Thanks - is that the correct Upwork profile? It doesn't look like it matches!

**Comment by conorpendergrast** [comment]
Payouts due:

- [ ] Issue Reporter: $250 @Yokabdk
- [ ] Contributor: $1000 + 50% urgency = $1500 @DanutGavrus 
- [ ] Contributor+: $1000 + 50% urgency = $1500 @mollfpr 

Eligible for 50% #urgency bonus? [Yes](https://github.com/Expensify/App/issues/22658#issuecomment-1651909745)

[Upwork job is here](https://www.upwork.com/jobs/~014ad96fffac1673e2)

**Comment by conorpendergrast** [comment]
@DanutGavrus Paid!

**Comment by Yokabdk** [comment]
@conorpendergrast yes it is i am using it so that i can get the payment easily

**Comment by conorpendergrast** [comment]
Yes, I saw that you said easily - can you add some more specifics on what you mean though?

**Comment by Yokabdk** [comment]
@conorpendergrast The upwork account I linked is my friend's account, since I was not able to verify my own upwork profile due to some personal issues I was thinking to accept the payment through his upwork account.

**Comment by conorpendergrast** [comment]
Ok! Offer sent now

**Comment by Yokabdk** [comment]
@conorpendergrast offer accepted

**Comment by conorpendergrast** [comment]
Paid! I'll do the regression test and will close this out today

**Comment by conorpendergrast** [comment]
Done! https://github.com/Expensify/Expensify/issues/306623
